File: linkedin1.py
import heapq

def lastStoneWeight(weights):
    for i in range(len(weights)):
        weights[i] *= -1
    heapq.heapify(weights)
    while weights and len(weights) > 1:
        print(weights)
        largest = heapq.heappop(weights)
        seclargest = heapq.heappop(weights)
        
        print(largest, seclargest)
        if largest == seclargest:
            continue
        else:
            largest -= seclargest
            heapq.heappush(weights, largest)
    if not weights:
        return 0
    return -1*weights[0]

print(lastStoneWeight([1,2,3,6,7,7]))

File: linkedin1.cpp
int lastStoneWeight(vector<int> &weights)
{
    priority_queue<int> pq(begin(weights), end(weights));
    while (pq.size() > 1)
    {
        int x = pq.top();
        pq.pop();
        int y = pq.top();
        pq.pop();
        if (x > y)
            pq.push(x - y);
    }
    return pq.empty() ? 0 : pq.top();
}